jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -445,11 +445,19 @@ private static Class<?> forName(String className, Class<?> caller) throws ClassN
 	if (null == sm) {
 		if (null != caller) {
 			callerClassLoader = caller.internalGetClassLoader();
+		/*[IF JAVA_SPEC_VERSION >= 19]*/
+		} else {
+			callerClassLoader = ClassLoader.internalGetSystemClassLoader();
+		/*[ENDIF] JAVA_SPEC_VERSION >= 19 */
 		}
 		return forNameImpl(className, true, callerClassLoader);
 	}
 	if (null != caller) {
 		callerClassLoader = caller.getClassLoaderImpl();
+	/*[IF JAVA_SPEC_VERSION >= 19]*/
+	} else {
+		callerClassLoader = ClassLoader.internalGetSystemClassLoader();
+	/*[ENDIF] JAVA_SPEC_VERSION >= 19 */
```

Comment:
I would suggest following to reduce some code duplication:
```suggestion
	ClassLoader callerClassLoader = null;
	if (caller != null) {
		callerClassLoader = caller.internalGetClassLoader();
	}
/*[IF JAVA_SPEC_VERSION >= 19]*/
	else {
		callerClassLoader = ClassLoader.internalGetSystemClassLoader();
	}
/*[ENDIF] JAVA_SPEC_VERSION >= 19 */
	if (sm == null) {
		return forNameImpl(className, true, callerClassLoader);
	}
```
`caller.getClassLoaderImpl()` might return `bootstrapClassLoader` instead of `null` which will be converted to `vm->systemClassLoader` w/o `J9VMJAVALANGCLASSLOADER_VMREF`.

Reply:
I don't think this suggestion is appropriate. `internalGetClassLoader()` will answer `null` instead of `bootstrapClassLoader` which would then be mapped to `vm->systemClassLoader` - a different behavior than we had previously.

Reply:
`bootstrapClassLoader` underhood is `vm->systemClassLoader`.

Reply:
It seems fine, but how about I do it in a separate PR.
The code was moved around by https://github.com/eclipse-openj9/openj9/pull/14267, which kept the original behaviour, but it can be refactored.

Reply:
+1: separate PR

Reply:
Jason, did you want to open a PR (considering you suggested the change) or shall I do it? Your suggestion looks fine to me.

Reply:
@pshipton either way is fine, technically that's a followup of this PR.

Reply:
Created https://github.com/eclipse-openj9/openj9/pull/15343